moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -576,7 +595,70 @@ bool ServoCalcs::cartesianServoCalcs(geometry_msgs::msg::TwistStamped& cmd,
   Eigen::MatrixXd matrix_s = svd.singularValues().asDiagonal();
   Eigen::MatrixXd pseudo_inverse = svd.matrixV() * matrix_s.inverse() * svd.matrixU().transpose();
 
-  delta_theta_ = pseudo_inverse * delta_x;
+  // Convert from cartesian commands to joint commands
+  // Use an IK solver plugin if we have one, otherwise use inverse Jacobian.
+  if (!use_inv_jacobian_)
+  {
+    // get a transformation matrix with the desired position change &
+    // get a transformation matrix with desired orientation change
+    Eigen::Isometry3d tf_pos_delta(Eigen::Isometry3d::Identity());
+    tf_pos_delta.translate(Eigen::Vector3d(delta_x[0], delta_x[1], delta_x[2]));
+
+    Eigen::Isometry3d tf_rot_delta(Eigen::Isometry3d::Identity());
+    Eigen::Quaterniond q = Eigen::AngleAxisd(delta_x[3], Eigen::Vector3d::UnitX()) *
+                           Eigen::AngleAxisd(delta_x[4], Eigen::Vector3d::UnitY()) *
+                           Eigen::AngleAxisd(delta_x[5], Eigen::Vector3d::UnitZ());
+    tf_rot_delta.rotate(q);
+
+    // Poses passed to IK solvers are assumed to be in the tip link (EE) reference frame
```

Comment:
I think these comments ought to be more generic. It's not necessarily the tip link reference frame, but rather whatever the IK reference frame is set to, right?

Otherwise the feature of switching from World to EE frame as the user's frame of reference (mentioned in [the tutorial](https://moveit.picknik.ai/humble/doc/examples/realtime_servo/realtime_servo_tutorial.html)) would not work.

Reply:
Yeah, it really should be more generic -- but it's actually more than just the comment. Really, we shouldn't just blindly use `tf_moveit_to_ee_frame_`, but rather a transform from what the IK solver knows as the base frame -> what the IK solver knows as the tip frame (which is likely the same as what servo knows as moveit -> ee frame, but not necessarily). 

Switching of the reference frame for the incoming twist commands is already handled above in the same function, so either way it should work for any incoming frame of reference.

See the latest commit for my solution to this